### PR TITLE
add InsecureSkipServerNameVerify to tls.Config

### PR DESCRIPTION
**Pull request by TNQOYxNU**

### CODE REVIEW

File: handshake_client.go [hunk 1]
```diff
@@ -40,9 +40,13 @@ var testingOnlyForceClientHelloSignatureAlgorithms []SignatureScheme
 
 func (c *Conn) makeClientHello() (*clientHelloMsg, ecdheParameters, error) {
 	config := c.config
-	if len(config.ServerName) == 0 && !config.InsecureSkipVerify {
+
+	// [UTLS SECTION START]
+	skipServerNameVerify := config.InsecureSkipVerify || config.InsecureSkipServerNameVerify
+	if len(config.ServerName) == 0 && !skipServerNameVerify {
 		return nil, nil, errors.New("tls: either ServerName or InsecureSkipVerify must be specified in the tls.Config")
```
Comment (gaukas) [inline handshake_client.go:47]: Could you please also update the error message to reflect your changes made to `Config`

File: u_conn.go [hunk 2]
```diff
@@ -377,7 +377,8 @@ func (c *UConn) clientHandshake(ctx context.Context) (err error) {
 	// [uTLS section begins]
 	// don't make new ClientHello, use hs.hello
 	// preserve the checks from beginning and end of makeClientHello()
-	if len(c.config.ServerName) == 0 && !c.config.InsecureSkipVerify {
+	skipServerNameVerify := c.config.InsecureSkipVerify || c.config.InsecureSkipServerNameVerify
+	if len(c.config.ServerName) == 0 && !skipServerNameVerify {
 		return errors.New("tls: either ServerName or InsecureSkipVerify must be specified in the tls.Config")
```
Comment (gaukas) [inline u_conn.go:382]: This line as well

**Review by gaukas — CHANGES_REQUESTED** [review]

**Comment by gaukas** [comment]
Also `go test` has failed. Please fix referring to the error message below: 
```
=== RUN   TestCloneNonFuncFields
    tls_test.go:834: all fields must be accounted for, but saw unknown field "InsecureSkipServerNameVerify"
```

**Review by gaukas — APPROVED** [review]

**Comment by gaukas** [comment]
Thanks! I will now get this merged.

**Comment by RPRX** [comment]
Nice work! But it would be better that if we can indicate another ServerName to verify, see https://github.com/refraction-networking/utls/pull/161

**Pull request merged**